src/init.cpp

```diff
@@ -1807,6 +1814,21 @@ bool AppInitMain(NodeContext& node)
         return false;
     }
 
+    // set asmap if supplied
+    if (gArgs.IsArgSet("-asmap")) {
+        const fs::path asmap_path = GetDataDir() / gArgs.GetArg("-asmap", DEFAULT_ASMAP_FILENAME);
+        std::vector<bool> asmap = CAddrMan::DecodeAsmap(asmap_path);
+        if (asmap.size() == 0) {
+            InitError(strprintf(_("Could not parse specified asmap: '%s'").translated, asmap_path));
```

Comment:
Further to https://github.com/bitcoin/bitcoin/pull/16702#discussion_r356739537, would suggest here `"Could not find or parse specified asmap: '%s'"`.

Rationale: the most frequent cause is likely to be an incorrect path or filename, not an unparseable file, and this error message will be output in both cases.